src/cli/add.py

```diff
@@ -16,8 +24,32 @@ def add(identifier):
         lockfile.
 
         If your project is a CurseForge modpack, IDENTIFIER is the CurseForge
-        project ID.
+        file ID.
 
         If your project is a Modrinth modpack, IDENTIFIER is the Modrinth
         project slug.
         """
+        with M3ContextManager() as context:
+            if context.config is None or context.lockfile is None:
+                raise click.ClickException('Not an m3 project')
+            if identifier.isdigit():
```

Comment:
We will probably have our own custom identifiers to show for `m3 ls` to use, which should be of the form

`[a-z][a-z\-]+[a-z]`

Reply:
or something like that

Reply:
Yes, the implementation and tests here will be updated when we implement the logic for constructing custom identifiers.